app/frontend/containers/CallToolView/CallToolView.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import _ from 'lodash';
+import ChampaignAPI from '../../util/ChampaignAPI';
+import type { OperationResponse } from '../../util/ChampaignAPI';
+
+import Form from '../../components/CallTool/Form';
+
+export type Target = {
+  countryCode: string;
+  name: string;
+  title: string;
+}
+
+export type Country = {
+  code: string;
+  name: string;
+}
+
+type OwnState = {
+  form: {
+    memberPhoneNumber?: string;
+    countryCode?: string;
+  };
+  errors: {
+    memberPhoneNumber?: any;
+    countryCode?: any;
+    base?: any[];
+  };
+  loading: boolean;
+  selectedTarget?: Target;
+}
+
+type OwnProps = {
+  memberPhoneNumber?: string;
+  countryCode?: string;
+  title?: string;
+  pageId: string | number;
+  targets: Target[];
+  targetCountries: Country[];
+}
+
+class CallToolView extends Component {
+  state: OwnState;
+  props: OwnProps;
+
+  constructor(props: OwnProps) {
+    super(props);
+    this.state = {
+      form: {},
+      errors: {},
+      loading: false
+    };
+  }
+
+  countryCodeChanged(countryCode: string) {
+    this.setState({
+      form: Object.assign({}, this.state.form, {countryCode: countryCode}),
```

Comment:
For future reference, setState also [accepts a function](https://facebook.github.io/react/docs/react-component.html#setstate), which is the safe way of accessing previous state:

```js
this.setState((prevState, props) => {
  return {
    form: { ...prevState.form, { countryCode } },
    ...
  };
});